onnx/onnx.in.proto

```diff
@@ -135,6 +202,20 @@ message AttributeProto {
   repeated bytes strings = 9;         // list of UTF-8 strings
   repeated TensorProto tensors = 10;  // list of tensors
   repeated GraphProto graphs = 11;    // list of graph
+
+// #if ONNX-ML
+  optional TupleAttribute        tuple    = 100; // A single tuple value
+  optional RecordAttribute       record   = 101; // A single record value
+  optional NullableAttribute     nullable = 102; // A single nullable value
+  optional SparseTensorAttribute sparse   = 103; // A single sparse tensor
+  optional TypeProto             typea    = 104; // A single type
```

Comment:
Record type does not seem nominal yet. If we do make it nominal, would we want to typically pass the name of the record-type (a string) here? Or, are there use-cases where we would need a full-fledged "TypeProto"?

Reply:
We'll follow the string encoding that Yuan proposed, I think.